common/datablocks/src/data_block.rs

```diff
@@ -254,8 +257,20 @@ impl fmt::Debug for DataBlock {
     }
 }
 
+impl Clone for DataBlock {
+    fn clone(&self) -> Self {
+        Self::create(self.schema.clone(), self.columns.clone())
```

Comment:
Do we need `super::metrics::incr_data_block_created();`?

Reply:
this clone() calls the `create` method, which contains a `incr_data_block_created()` inside 🤔